diff --git a/src/tds/tls.c b/src/tds/tls.c
index 82ae4c2..cc226fb 100644
--- a/src/tds/tls.c
+++ b/src/tds/tls.c
@@ -940,7 +940,7 @@ int
 tds_ssl_init(TDSSOCKET *tds)
 {
 #define DEFAULT_OPENSSL_CTX_OPTIONS (SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3 | SSL_OP_NO_TLSv1)
-#define DEFAULT_OPENSSL_CIPHERS "HIGH:!SSLv2:!aNULL:-DH"
+#define DEFAULT_OPENSSL_CIPHERS "PROFILE=SYSTEM"
 
 	SSL *con;
 	SSL_CTX *ctx;
